doc/user/ospf6d.rst

```diff
@@ -198,10 +198,135 @@ OSPF6 interface
 
    Sets interface's Inf-Trans-Delay. Default value is 1.
 
-.. clicmd:: ipv6 ospf6 network (broadcast|point-to-point)
+.. clicmd:: ipv6 ospf6 network (broadcast|point-to-point|point-to-multipoint)
 
    Set explicitly network type for specified interface.
 
+   The only functional difference between ``point-to-point`` (PtP) and
+   ``point-to-multipoint`` (PtMP) mode is the packet addressing for database
+   flooding and updates.  PtP will use multicast packets while PtMP will
+   unicast them.  Apart from this,
+   :clicmd:`ipv6 ospf6 p2p-p2mp connected-prefixes <include|exclude>` has a
+   different default for PtP and PtMP.  There are no other differences, in
+   particular FRR does not impose a limit of one neighbor in PtP mode.
+
+   FRR does not support NBMA mode for IPv6 and likely never will, as NBMA is
+   considered deprecated for IPv6.  Refer to `this IETF OSPF working group
+   discussion
+   <https://mailarchive.ietf.org/arch/msg/ospf/8GAbr4qSMMt5J7SvAcZQ1H7ARhk/>`_
+   for context.
+
+OSPF6 point-to-point and point-to-multipoint operation
+======================================================
+
+OSPFv3, by default, operates in broadcast mode where it elects a DR and BDR
+for each network segment.  This can be changed to point-to-point (PtP) /
+point-to-multipoint (PtMP) mode by configuration.  The actual physical
+interface characteristics do not matter for this setting, all interfaces can
+be configured for all modes.  However, routers must be configured for the same
+mode to form adjacencies.
+
+The main advantages of PtP/PtMP mode are:
+
+- no DR/BDR election
+- adjacencies can be suppressed in a pairwise manner for any two routers, e.g.
+  to represent the underlying topology if it isn't a true full mesh
+- distinct costs can be set for each pair of routers and direction
+
+The main downside is less efficient flooding on networks with a large number
+of OSPFv3 routers.
```

Comment:
There's actually another downside, but I don't know if it's worth going into in the comments. :-) With the DR, all the routers on a single broadcast interface look like a hub-and-spoke rather than a full mesh. This reduces the size of the spt, so spf can calculate just a hair faster. It's not a "thing" in most networks and on most processors now, but way back in the day when all our server room floors were dirt, it made a difference. :-)

Reply:
That's kinda listed as an advantage because the very same thing allows you to control metrics in a n×m manner :smile: … but yeah that's another reason why I put the "advanced options, make sure you know what you're doing" above.